src/gleam/string.gleam

```diff
@@ -309,15 +309,17 @@ pub fn join(strings: List(String), with separator: String) -> String {
 //
 //
 // pub fn pad_right(string: String, to size: Int, with: String) {}
-// TODO
+
 // Get rid of whitespace on both sides of a String.
 //
 // ## Examples
 //    > trim("  hats  \n")
 //    "hats"
 //
 //
-// pub fn trim(string: String) -> String {}
+pub external fn trim(String) -> String =
+  "gleam_stdlib" "string_trim"
+
```

Comment:
Could we take an approach similar to this function? https://github.com/gleam-lang/stdlib/blob/8069fabc19c23197d551d0f969a135840d86d808/src/gleam/iodata.gleam#L99-L110

To keep as much in Gleam as possible.

Reply:
To be clear I would like to not have any new functions in the gleam_stdlib.erl module, instead import the `string:trim/2` function and pass in a custom type `Both`

Reply:
Ah, I see. I'll have a look, thanks.

Reply:
Ok, I understand now what's going on here, this is a type safe mapping to an Erlang atom. And adding to `gleam_stdlib.erl` was unnecessary because no further logic is required on the Erlang side.
To my untrained eyes it looks okay now.